common/darwin/Classes/FlutterRTCPeerConnection.m

```diff
@@ -388,19 +390,25 @@ - (void)peerConnection:(RTCPeerConnection *)peerConnection didAddStream:(RTCMedi
     NSMutableArray *audioTracks = [NSMutableArray array];
     NSMutableArray *videoTracks = [NSMutableArray array];
 
+    BOOL hasAudio = NO;
     for (RTCAudioTrack *track in stream.audioTracks) {
         peerConnection.remoteTracks[track.trackId] = track;
         [audioTracks addObject:@{@"id": track.trackId, @"kind": track.kind, @"label": track.trackId, @"enabled": @(track.isEnabled), @"remote": @(YES), @"readyState": @"live"}];
+        hasAudio = YES;
     }
-    
+
     for (RTCVideoTrack *track in stream.videoTracks) {
         peerConnection.remoteTracks[track.trackId] = track;
         [videoTracks addObject:@{@"id": track.trackId, @"kind": track.kind, @"label": track.trackId, @"enabled": @(track.isEnabled), @"remote": @(YES), @"readyState": @"live"}];
     }
-    
+
     NSString *streamId = stream.streamId;
     peerConnection.remoteStreams[streamId] = stream;
-    
+
+    if (hasAudio) {
+        [AudioUtils ensureAudioSessionWithRecording:NO];
```

Comment:
When `didAddStream` was triggered, we received a `remote track`, but before that, we had created a `local track` to capture audio. Is there any wrong result here?

Reply:
that case works fine, `ensureAudioSessionWithRecording` will always upgrade the audio session, and will never downgrade.

* when we received a track: sets to Playback
* then create local track: sets to PlayAndRecord
* then received another track: does nothing

Reply:
I think there is an easier way:

```
@interface FlutterWebRTCPlugin : NSObject<FlutterPlugin, RTCPeerConnectionDelegate>
...

-(void) ensureAudioSessionWithRecording {
  BOOL hasRecordingTrack = NO;
  for (RTCMediaStreamTrack *track in self.localTracks) {
         if([track isKindOfClass:[RTCAudioTrack class]]) {
                hasRecordingTrack = YES;
         }
   }

 BOOL hasPlaybackTrack = NO;
 for(RTCPeerConnection *pc in self.peerConnections) {
          for (RTCMediaStreamTrack *track in pc.remoteTracks) {
             if([track isKindOfClass:[RTCAudioTrack class]]) {
                    hasPlaybackTrack = YES;
             }
      }
 }
 if(hasRecordingTrack ... hasPlaybackTrack) {
     //do something here.
  }
}
```

Reply:
sure, that works too!

Reply:
Then the `ensureAudioSessionWithRecording` method is called after the number of any remote/local track changes.

Reply:
@davidzhao What do you think about it?

Reply:
sure, I can update it